code/datums/components/embedded.dm

```diff
@@ -124,81 +131,105 @@
-
+		victim.visible_message("<span class='danger'>[weapon] sticks itself to [victim]'s [limb.name]!</span>",ignored_mobs=victim)
+		to_chat(victim, "<span class='userdanger'>[weapon] sticks itself to your [limb.name]!</span>")
 
 /// Called every time a human with a harmful embed moves, rolling a chance for the item to cause pain. The chance is halved if the human is crawling or walking.
 /datum/component/embedded/proc/jostleCheck()
-	var/mob/living/carbon/human/victim = parent
+	var/mob/living/carbon/victim = parent
 
 	var/chance = jostle_chance
 	if(victim.m_intent == MOVE_INTENT_WALK || !(victim.mobility_flags & MOBILITY_STAND))
 		chance *= 0.5
 
 	if(harmful && prob(chance))
 		var/damage = weapon.w_class * jostle_pain_mult
-		L.receive_damage(brute=(1-pain_stam_pct) * damage, stamina=pain_stam_pct * damage)
-		to_chat(victim, "<span class='userdanger'>[weapon] embedded in your [L.name] jostles and stings!</span>")
+		limb.receive_damage(brute=(1-pain_stam_pct) * damage, stamina=pain_stam_pct * damage)
+		to_chat(victim, "<span class='userdanger'>[weapon] embedded in your [limb.name] jostles and stings!</span>")
 
 
 /// Called when then item randomly falls out of a human. This handles the damage and descriptors, then calls safe_remove()
 /datum/component/embedded/proc/fallOutHuman()
-	var/mob/living/carbon/human/victim = parent
+	var/mob/living/carbon/victim = parent
 
 	if(harmful)
 		var/damage = weapon.w_class * remove_pain_mult
-		L.receive_damage(brute=(1-pain_stam_pct) * damage, stamina=pain_stam_pct * damage)
-		victim.visible_message("<span class='danger'>[weapon] falls out of [victim.name]'s [L.name]!</span>","<span class='userdanger'>[weapon] falls out of your [L.name]!</span>")
+		limb.receive_damage(brute=(1-pain_stam_pct) * damage, stamina=pain_stam_pct * damage)
+		victim.visible_message("<span class='danger'>[weapon] falls out of [victim.name]'s [limb.name]!</span>", ignored_mobs=victim)
+		to_chat(victim, "<span class='userdanger'>[weapon] falls out of your [limb.name]!</span>")
 	else
-		victim.visible_message("<span class='danger'>[weapon] falls off of [victim.name]'s [L.name]!</span>","<span class='userdanger'>[weapon] falls off of your [L.name]!</span>")
+		victim.visible_message("<span class='danger'>[weapon] falls off of [victim.name]'s [limb.name]!</span>", ignored_mobs=victim)
+		to_chat(victim, "<span class='userdanger'>[weapon] falls off of your [limb.name]!</span>")
 
 	safeRemoveHuman()
 
 
 /// Called when a human with an object embedded/stuck to them inspects themselves and clicks the appropriate link to begin ripping the item out. This handles the ripping attempt, descriptors, and dealing damage, then calls safe_remove()
-/datum/component/embedded/proc/ripOutHuman()
-	var/mob/living/carbon/human/victim = parent
+/datum/component/embedded/proc/ripOutHuman(datum/source, obj/item/I, obj/item/bodypart/limb)
+	if(I != weapon || limb != limb)
```

Comment:
```suggestion
	if(I != weapon || src.limb != limb)
```
When a local variable has the same name than an input variable, that's when you use the `src.` prefix. Else you are comparing the variable to itself.
If you mean to reference the `obj/item/bodypart/limb` that's being input in the proc, use `limb`.
If you want to reference the `var/obj/item/bodypart/limb` defined for the component, use `src.limb`.

Reply:
woopsie, the chaos of mass replace